File: app/src/main/java/com/cottondroid/todoachieved/task/model/ServerCreatedTimestampConverter.java
package com.cottondroid.todoachieved.task.model;

import android.arch.persistence.room.TypeConverter;

import java.util.HashMap;
import java.util.Map;

public class ServerCreatedTimestampConverter {
    private static final String SERVER_CREATED_TIMESTAMP_KEY = "serverCreatedTimestamp";

    @TypeConverter
    public static Map<String, String> toServerTimestamp(Long value) {
        if (value == null) {
            return null;
        }
        Map<String, String> timestampMap = new HashMap<>();
        timestampMap.put(SERVER_CREATED_TIMESTAMP_KEY, String.valueOf(value));
        return timestampMap;
    }

    @TypeConverter
    public static Long fromServerTimestamp(Map<String, String> serverTimestamp) {
        try {
            return Long.valueOf(serverTimestamp.get(SERVER_CREATED_TIMESTAMP_KEY));
        } catch (Exception e) {
            return null;
        }
    }
}

File: app/src/main/java/com/cottondroid/todoachieved/task/model/ServerCreatedTimestampConverter.kt
package com.cottondroid.todoachieved.task.model

import androidx.room.TypeConverter

class ServerCreatedTimestampConverter {

    companion object {
        private const val SERVER_CREATED_TIMESTAMP_KEY = "serverCreatedTimestamp"

        @JvmStatic
        @TypeConverter
        fun toServerCreatedTimestamp(value: Long?): Map<String, String>? {
            if (value == null) {
                return null
            }
            val timestampMap: MutableMap<String, String> = HashMap()
            timestampMap[SERVER_CREATED_TIMESTAMP_KEY] = value.toString()
            return timestampMap
        }

        @JvmStatic
        @TypeConverter
        fun fromServerCreatedTimestamp(serverTimestamp: Map<String?, String?>): Long? {
            return serverTimestamp[SERVER_CREATED_TIMESTAMP_KEY]?.toLong()
        }
    }
}